frontend/pkg/frontend/middleware_tracing.go

```diff
@@ -0,0 +1,28 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func MiddlewareTracing(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	attrs := []attribute.KeyValue{}
+	for k, v := range r.Header {
+		attrs = append(attrs, attribute.String(fmt.Sprintf("http.request.header.%s", k), strings.Join(v, ",")))
```

Comment:
Are we sure that it's ok to report all of the incoming HTTP headers? E.g. any personal/private/sensitive information which shouldn't be reported?
Maybe it's better to start with an empty set of custom attributes?